pandas/tests/indexes/test_category.py

```diff
@@ -598,17 +598,32 @@ def test_reindex_dtype(self):
         tm.assert_index_equal(res, exp, exact=True)
         tm.assert_numpy_array_equal(indexer, np.array([0, 3, 2], dtype=np.intp))
 
-    def test_reindex_duplicate_target(self):
+    def test_reindex_duplicate_source(self):
         # See GH23963
         c = CategoricalIndex(["a", "b", "c", "a"], categories=["a", "b", "c", "d"])
-        with pytest.raises(ValueError, match="non-unique indexer"):
-            c.reindex(["a", "a", "c"])
+        with pytest.raises(ValueError, match="duplicate axis"):
+            c._can_reindex(["a", "c"])
 
-        with pytest.raises(ValueError, match="non-unique indexer"):
-            c.reindex(
-                CategoricalIndex(["a", "a", "c"], categories=["a", "b", "c", "d"])
+        with pytest.raises(ValueError, match="duplicate axis"):
+            c._can_reindex(
+                CategoricalIndex(["a", "c"], categories=["a", "b", "c", "d"])
             )
 
+    def test_reindex_duplicate_target(self):
+        # See GH25459
+        c = CategoricalIndex(["a", "b", "c"], categories=["a", "b", "c", "d"])
```

Comment:
i have a pet peeve about 1-letter variable names.  can you make this e.g. `cat`.  same above

Reply:
OK, but please note that this renders my new tests inconsistent with the surrounding ones.